handbook/handbook.sop.git-workflow.md

```diff
@@ -0,0 +1,86 @@
+---
+id: tIFX0R6iEvZUGkOJluYo1
+title: Git Workflow
+desc: ''
+updated: 1624216920883
+created: 1624215795020
+---
+
+## Summary
+
+How we work with code and git
+
+## Process
+
+We employ the [feature branch workflow](https://www.atlassian.com/git/tutorials/comparing-workflows/feature-branch-workflow) when working with git. 
+
+### Working on a new Feature
+This means that every feature will go into a feature branch during local development and pushed to the origin with the same name. When changes are ready to be reviewed, submit a pull request against `master`. 
+
+For feature branches, we use the following conventions:
+- feat/{feature-name}
+- fix/{fix-name}
+
+### Commit Style
+- we follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) with the following tags
+- categories:
+  - feat: feature - introduce new functionality
+  - enhance: enhancement - improve existing functionality
+  - fix: make something not broken
+  - chore: backend improvements
+  - spike: not complete commit
+ 
+If you are introducing a breaking change, either add an `!` after the category or a footer at the end of your message
+
+```
+feat: allow provided config object to extend other configs
+
+BREAKING CHANGE: `extends` key in config file is now used for extending other config files
+```
+> Example from [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) licensed under [CC BY 3.0](https://creativecommons.org/licenses/by/3.0/)
+
+```
+refactor!: drop support for Node 6
+```
+> Example from [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) licensed under [CC BY 3.0](https://creativecommons.org/licenses/by/3.0/)
+
+### Pre-Requisites
+
+1. Make sure there's a [[test|dendron.dev.qa#writing-tests]] for the new feature
+1. Make sure all the existing [[tests|dendron.dev.qa#running-all-tests]] pass 
+1. Do a spot check by running your feature with our [[test Workspace|dendron.dev.qa#test-workspace]]
+1. If your change reflects documentation changes, also submit a PR to [dendron-site](https://github.com/dendronhq/dendron-site) and mention the doc PR link in your current PR
+
+
+### Creating a pull request
+
+Because we do weekly releases, we like to do a weekly cadence for pull requests as well. Pull requests for next week's release should be in 5pm local time (your timezone) on Thursday at the latest. 
```

Comment:
If it works for you, it might be clearer to go with [Thursday Anywhere on Earth](https://www.timeanddate.com/worldclock/converter.html?iso=20210625T115900&p1=tz_aoe&p2=234&p3=64&p4=805&p5=102).

It's before Friday morning for US and before Friday evening for Asia time zones, and it means you get the pull requests before Friday's work day.

Reply:
updated